presentation/otp/src/main/java/com/hilingual/presentation/otp/component/OtpTextField.kt

```diff
@@ -0,0 +1,173 @@
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.hilingual.core.designsystem.theme.HilingualTheme
+
+@Composable
+fun OtpTextField(
+    otpText: () -> String,
+    onOtpTextChange: (String) -> Unit,
+    modifier: Modifier = Modifier,
+    isError: Boolean = false
+) {
+    val otpValue = otpText()
+    val focusManager = LocalFocusManager.current
+    BasicTextField(
+        modifier = modifier,
+        value = otpValue,
+        onValueChange = {
+            if (it.length <= 6 && it.all { char -> char.isDigit() }) {
+                onOtpTextChange(it)
+            }
+        },
+        keyboardActions = KeyboardActions(
+            onDone = { focusManager.clearFocus() }
+        ),
+        keyboardOptions = KeyboardOptions(
+            keyboardType = KeyboardType.NumberPassword,
+            imeAction = ImeAction.Done
+        ),
+        decorationBox = {
+            Row(
+                horizontalArrangement = Arrangement.spacedBy(8.dp),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                repeat(6) { index ->
+                    OtpChar(
+                        char = otpValue.getOrNull(index),
+                        isError = isError
+                    )
+                }
+            }
+        }
+    )
+}
+
+@Composable
+private fun OtpChar(
+    char: Char?,
+    isError: Boolean
+) {
+    val isFilled = char != null
+
+    val borderColor = when {
+        isError -> HilingualTheme.colors.alertRed
+        isFilled -> HilingualTheme.colors.hilingualBlack
+        else -> Color.Transparent
+    }
+
+    val backgroundColor = when {
+        isError || isFilled -> HilingualTheme.colors.white
+        else -> HilingualTheme.colors.gray100
+    }
+
+    val textColor = if (isError) HilingualTheme.colors.alertRed else HilingualTheme.colors.black
+
+    Box(
+        modifier = Modifier
+            .size(width = 48.dp, height = 60.dp)
+            .clip(RoundedCornerShape(8.dp))
+            .background(backgroundColor)
+            .border(
+                width = 1.dp,
+                color = borderColor,
+                shape = RoundedCornerShape(8.dp)
+            ),
+        contentAlignment = Alignment.Center
+    ) {
+        Text(
+            text = char?.toString() ?: "",
```

Comment:
```suggestion
            text = char?.toString().orEmpty(),
```
P100: (Just Wonder) 엘비스 연산자 사용하는 방법과 orEmpty 사용하는 방법 중 어떤게 더 코틀린스러운 코드라고 생각하시나요?

Reply:
둘 다 코틀린 철학에 맞는 코드라고 생각합니다. 

`orEmpty()`가 있는데 엘비스 연산자를 사용하는 이유가 뭘까요?

**범용성 측면에서** 엘비스 연산자는 빈 문자열뿐만 아니라 다양한 기본값 설정이 가능합니다. 반면 `orEmpty()`는 null을 빈 값으로 변환하는 의도가 명확할 때 사용하면 좋습니다.

**성능적으로는** 엘비스 연산자가 근소하게 더 빠릅니다. 아주 미미한 차이지만, `orEmpty()`는 함수 호출 오버헤드가 있는 반면 엘비스 연산자는 컴파일러 레벨에서 직접 최적화됩니다.

**가독성 측면에서** 엘비스 연산자는 코틀린의 기본 문법이므로 모든 팀원이 즉시 이해할 수 있습니다. 함수 호출의 경우 처음 보는 개발자에게는 한 번에 알기 어려울 수 있죠. 물론 `orEmpty()`는 이제 기본적인 함수라고 생각합니다😅

결론적으로 **두 방식 모두 동등하게 코틀린스럽다**고 생각합니다. 상황과 팀 컨벤션에 따라 적절히 선택하면 될 것 같습니다!

@Hi-lingual/team-android 팀의 생각은 어떤지?